### ISSUE TITLE
Incorrect behaviour when returning array.

### ISSUE THREAD

**Issue by racinmat** [slevomat/coding-standard#182]
In my class, PlayerRepository, coding standard complaints about both missing type for array and having redundant phpdoc. 
Here is code which has the issue:
```php
<?php declare(strict_types=1);

namespace AppBundle\Repository;

use AppBundle\Entity\Game;
use AppBundle\Entity\Player;
use AppBundle\Entity\Team;
use Doctrine\ORM\EntityRepository;

class PlayerRepository extends BaseRepository {

	/**
	 * @return string[]
	 */
	public function getInvitablePlayerUsernames(Team $team): array {
		$qb = $this->getEntityManager()->createQueryBuilder();
		//...
		return array_column($qb->getQuery()->getResult(), 'username');
	}

}
```
And output of the coding standard:

 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  Line   src\AppBundle\Repository\PlayerRepository.php
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
  1      Method \AppBundle\Repository\PlayerRepository::getInvitablePlayerUsernames() does not have @return annotation for its traversable return value.
         (SlevomatCodingStandard\Sniffs\TypeHints\TypeHintDeclarationSniff)
  1      Method \AppBundle\Repository\PlayerRepository::getInvitablePlayerUsernames() does not need documentation comment.
         (SlevomatCodingStandard\Sniffs\TypeHints\TypeHintDeclarationSniff)
 ------ ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------

**Comment by kukulich** [comment]
Closing because of no response.

**Comment by racinmat** [comment]
I tried to reproduce this my bug, but with no luck, so there was probably some magic. It works now.
